src/commands/registries/azure/deployImageToAzure.ts

```diff
@@ -88,23 +87,22 @@ async function getNewSiteConfig(node: RemoteTagTreeItemBase): Promise<SiteConfig
                 password = nonNullProp(cred, 'passwords')[0].value;
             }
             appSettings.push({ name: "DOCKER_ENABLE_CI", value: 'true' });
-        } else if (registryTI instanceof PrivateRegistryTreeItem) {
-            const auth: AuthOptions = await registryTI.getAuth();
-            username = auth.username;
-            password = auth.password;
+        } else if (registryTI instanceof GenericDockerV2RegistryTreeItem) {
+            username = registryTI.cachedProvider.username;
+            password = await getRegistryPassword(registryTI.cachedProvider);
         } else {
             throw new RangeError(`Unrecognized node type "${registryTI.constructor.name}"`);
         }
-
-        if (!username || !password) {
-            throw new Error(`Failed to get credentials for registry "${registryTI.host}".`);
-        }
+    } else {
+        throw new RangeError(`Unrecognized node type "${registryTI.constructor.name}"`);
     }
 
-    appSettings.push({ name: "DOCKER_REGISTRY_SERVER_USERNAME", value: username });
-    appSettings.push({ name: "DOCKER_REGISTRY_SERVER_PASSWORD", value: password });
+    if (username && password) {
+        appSettings.push({ name: "DOCKER_REGISTRY_SERVER_USERNAME", value: username });
```

Comment:
Are these settings only needed for specific types of Registries?

Reply:
Only ones that need authentication. For example, there are public docker hub images. Plus you could always add these settings later yourself

Reply:
Or more likely - self hosted registries might exist that don't have auth setup yet